packages/hatchify-core/src/assembler/types.ts

```diff
@@ -15,36 +15,41 @@ import type {
 } from "../dataTypes/string"
 import type { FinalAttribute, PartialAttribute } from "../types"
 
-export interface PartialSchema {
+export type PartialAttributeRecord = Record<
+  string,
+  | PartialAttribute<
+      PartialNumberORM,
+      PartialNumberControlType,
+      number,
+      FinalNumberORM
+    >
+  | PartialAttribute<
+      PartialStringORM,
+      PartialStringControlType,
+      string,
+      FinalStringORM
+    >
+  | PartialAttribute<
+      PartialDatetimeORM,
+      PartialDatetimeControlType,
+      Date,
+      FinalDatetimeORM
+    >
+>
+
+export interface PartialSchema<
+  TAttributes extends PartialAttributeRecord = PartialAttributeRecord,
```

Comment:
@roy-coder 
we had to convert this to a generic for the frontend - I think in the near future (when we do enums) we will have to modify `FinalSchema` as well, but we will cross that bridge when we get there 😄